samples/dotnet/EdgeX509AuthDownstreamDevice/Program.cs

```diff
@@ -152,27 +154,48 @@ internal static (X509Certificate2, IEnumerable<X509Certificate2>) ParseCertifica
             return ParseCertificateAndKey(cert, privateKey);
         }
 
-        static TransportType GetTransport(string protocol)
+        static ITransportSettings[] GetTransport(string protocol)
         {
-            TransportType result = TransportType.Mqtt_Tcp_Only;
+            TransportType transportType = TransportType.Mqtt_Tcp_Only;
+            ITransportSettings[] transportSettings = new ITransportSettings[1];
 
-            if (!string.IsNullOrWhiteSpace(protocol))
+            if (string.Compare("Mqtt", protocol, StringComparison.OrdinalIgnoreCase) == 0)
             {
-                if (string.Compare("MqttWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Mqtt_WebSocket_Only;
-                }
-                else if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Amqp_Tcp_Only;
-                }
-                else if (string.Compare("AmqpWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Amqp_WebSocket_Only;
-                }
+                transportType = TransportType.Mqtt_Tcp_Only;
+            }
+            else if (string.Compare("MqttWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Mqtt_WebSocket_Only;
+            }
+            else if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Amqp_Tcp_Only;
+            }
+            else if (string.Compare("AmqpWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Amqp_WebSocket_Only;
+            }
+            else
+            {
+                throw new ArgumentException("Invalid protocol");
+            }
+
+            X509Certificate2 trustedCACert = GetTrustedCACertFromFile(TrustedCACertPath);
+            RemoteCertificateValidationCallback certificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => CustomCertificateValidator.ValidateCertificate(trustedCACert, (X509Certificate2) certificate, chain, sslPolicyErrors);
+            if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0 || string.Compare("AmqpWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
nit: use transporType variable

Reply:
refer to https://github.com/Azure/iotedge/blob/584041aba7e20f7b21ffdf7e706f9d71da6006dc/smoke/LeafDevice/details/CustomCertificateValidator.cs#L41